listenbrainz_spark/tests/test_utils.py

```diff
@@ -1,14 +1,96 @@
+import os
 import unittest
+from datetime import datetime
 
-from listenbrainz_spark import utils
+import listenbrainz_spark
+from listenbrainz_spark import utils, hdfs_connection
 from listenbrainz_spark.tests import SparkTestCase
 
+from pyspark.sql import Row
+
+HDFS_CLUSTER_URI = 'hdfs://hadoop-master:9000'
+
 class UtilsTestCase(SparkTestCase):
 
+    def test_append_dataframe(self):
+        path_ = 'test_df.parquet'
+        hdfs_path = os.path.join(HDFS_CLUSTER_URI, path_)
+
+        df = utils.create_dataframe(Row(column1=1, column2=2), schema=None)
+        utils.append(df, hdfs_path)
+        new_df = utils.read_files_from_HDFS(hdfs_path)
+        self.assertEqual(new_df.count(), 1)
+
+        df = utils.create_dataframe(Row(column1=3, column2=4), schema=None)
+        utils.append(df, hdfs_path)
+        appended_df = utils.read_files_from_HDFS(hdfs_path)
+        self.assertEqual(appended_df.count(), 2)
+        # delete_dir dir in every test func so that using identical dirs by chance will not throw error.
```

Comment:
this should be a part of the `tearDown` function so that it automatically gets called after each test and we don't need to duplicate code.